std/random.d

```diff
@@ -359,12 +359,11 @@ $(D x0).
 /**
    (Re)seeds the generator.
 */
-    void seed(UIntType x0 = 1) @safe pure
+    void seed(UIntType x0 = 1) @safe @nogc pure nothrow
     {
         static if (c == 0)
         {
-            import std.exception : enforce;
-            enforce(x0, "Invalid (zero) seed for "
+            x0 || assert(0, "Invalid (zero) seed for "
```

Comment:
Yes. They're a special case. They're used when they need to fail even in release mode, which is _not_ the normal thing to do with assertions. It's for stuff like the default case of switch statement when it should never be hit or the catch statement that catches an exception that should never be thrown, but you have to catch it to make the function `nothrow`. It's not for validating function arguments.